client/components/upgrades/google-apps/google-apps-dialog/users-form.jsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormTextInput from 'components/forms/form-text-input';
+import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
+import FormInputValidation from 'components/forms/form-input-validation';
+
+class GoogleAppsUsersForm extends React.Component {
+	fieldClasses( fieldName ) {
+		return `google-apps-dialog__user-field google-apps-dialog__user-${ fieldName }`;
+	}
+
+	updateInputField = ( event ) => {
+		this.props.updateField( this.props.index, event );
+	};
+
+	recordInputFocusEmail = () => {
+		this.props.recordInputFocus( this.props.index, 'Email' );
+	};
+
+	recordInputFocusFirstName = () => {
+		this.props.recordInputFocus( this.props.index, 'First Name' );
+	};
+
+	recordInputFocusLastName = () => {
+		this.props.recordInputFocus( this.props.index, 'Last Name' );
+	};
+
+	render() {
+		const { index, user, domain, onBlur, translate } = this.props;
+		const contactText = translate(
+			'contact',
+			{
+				context: 'part of e-mail address',
+				comment: 'As it would be part of an e-mail address contact@example.com'
+			}
+		);
+
+		return (
+			<div className="google-apps-dialog__user-fields" key={ `google-apps-dialog-user-${ index }` }>
+				<FormFieldset>
+					<FormTextInputWithAffixes
+						className={ this.fieldClasses( 'email' ) }
+						placeholder={ translate( 'e.g. %(example)s', { args: { example: contactText } } ) }
+						name="email"
+						value={ user.email.value }
```

Comment:
Unless we specify that an object is required, note that it's generally a good idea to guard against multiple property accessors (`foo.bar.baz`) by checking for a truthy value or using lodash `get`

A common error is that an object isn't available yet (`undefined` or `null`), which may lead to a js error.